include/sdf/parser.hh

```diff
@@ -115,6 +115,21 @@ namespace sdf
   SDFORMAT_VISIBLE
   bool readFile(const std::string &_filename, SDFPtr _sdf);
 
+  /// \brief Populate the SDF values from a file
+  ///
+  /// This populates the given sdf pointer from a file. If the file is a URDF
+  /// file it is converted to SDF first. All files are converted to the latest
+  /// SDF version
+  /// \param[in] _filename Name of the SDF file
+  /// \param[in] _sdf Pointer to an SDF object.
+  /// \param[in] _enable_new_warnings Consider previous issues in debug log as
+  ///            warnings. This allow to keep log level backwards compatible
+  ///            while providing an option to increase the warnings with new
+  ///            relevant issues.
+  /// \return True if successful.
+  bool readFile(const std::string &_filename, SDFPtr _sdf,
```

Comment:
While I think this is super helpful, I'm not sure adding a new overload is the best option. In Citadel, we only have two existing overloads, so this adds a third. In Garden, we have 6 overloads already! and how this will interact with all of them when forward ported is not clear. Starting from Fortress, these functions take a `ParserConfig` object, so adding it to that makes sense, but for citadel, I would prefer if we could use a different approach. Perhaps an env variable? Or backport` ParserConfig`?

Reply:
before handling `sdf9`, I'd like to review a pull request that targets `sdf12` and uses `ParserConfig`. Once we have completed that review, we can decide how we want to backport it to `sdf9`. Does that sound ok?

Reply:
sure, I'll try to create it in the next days.

Reply:
Hello @azeey, @j-rivero, @scpeters! Circling back to this discussion, I've implemented the ideas for `sdf12`, https://github.com/gazebosim/sdformat/pull/1238,
* changing `sdfdbg` to `sdfwarn` whenever links are ignored during urdf->sdf conversion
* adding a setting in `ParserConfig` to convert urdf links with no inertia or masses smaller than 1e-6 to frames

I found that using warnings directly do not trigger any regressions, users will however start seeing them in their workflows without any change in behavior (might be a good thing, since these links are most likely mistakes)

Let me know what yall think!

Reply:
I think converting the debug messages to warnings (without any config flags) is reasonable since my expectation is that users generally do not rely on this behavior being the normal, i.e, we want to avoid spamming users who expect links to be ignored and they're okay with it, but I don't think that's most users.

Reply:
We just got https://github.com/gazebosim/sdformat/issues/199 merged into `sdf12`, changing debug messages to warnings, with more explicit information if links or joints were ignored, and suggestions to resolve those issues. There's no behavior change (but with 1 bug fix), and `ParserConfig` was not used. If it looks ok, we can probably try to cherry pick that back to `sdf9` for gazebo11 and citadel?

Reply:
@j-rivero @azeey https://github.com/gazebosim/sdformat/pull/1258 is the cherry-pick towards `sdf9`, it looks like the new warnings are not causing any regressions. Do yall think we still need an option to display newer warnings?